### PR TITLE
Filter by list added exactmatch

### PR DESCRIPTION
**Pull request by teizenman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20160

## Description
Added `exact match` argument to find specific values, while before that every substring was found

### CODE REVIEW

File: Scripts/FilterByList/FilterByList.yml [hunk 1]
```diff
@@ -1,37 +1,64 @@
...
+  name: ignorecase
+  predefined:
+  - 'yes'
+  - 'no'
+  required: false
+  secret: false
+- auto: PREDEFINED
+  default: false
+  defaultValue: 'No'
+  description: Whether to match the exact item in the list, ot look for any string
```
Comment (ronykoz) [inline Scripts/FilterByList/FilterByList.yml]: No and Yes are not intuitive options 
```suggestion
  description: Whether to match the exact item in the list, or look for any string
```

File: Scripts/FilterByList/FilterByList.py [hunk 2]
```diff
@@ -2,64 +2,84 @@
 from CommonServerPython import *
 from CommonServerUserPython import *
 
-lst = demisto.executeCommand("getList", {"listName": demisto.args()["listname"]})
 
-if isError(lst[0]):
-    demisto.results(lst)
-    sys.exit(0)
+def filter_list(lst, items, ignore_case, match_exact, list_name):
```
Comment (ronykoz) [inline Scripts/FilterByList/FilterByList.py]: split into several functions

File: Scripts/FilterByList/FilterByList.py [hunk 3]
```diff
@@ -2,64 +2,84 @@
...
-        hr += item + " is not part of the list\n"
-        notwhitelisted.append(item)
-
-ec = {"List.In": whitlisted, "List.NotIn": notwhitelisted}
-contents = {"inList": whitlisted, "notInList": notwhitelisted}
-demisto.results(
-    {
+
+    lst = lst[0]["Contents"].split(",")
+    search_flag = re.IGNORECASE if ignore_case else 0
```
Comment (ronykoz) [inline Scripts/FilterByList/FilterByList.py]: Not indicative name

**Comment by teizenman** [comment]
@ronykoz There is a TPB for the script. I didn't change it, only added UT

File: Scripts/FilterByList/FilterByList.py [hunk 4]
```diff
@@ -2,64 +2,97 @@
...
-    {
-        "ContentsFormat": formats["json"],
-        "Type": entryTypes["note"],
-        "Contents": contents,
-        "HumanReadable": hr,
-        "HumanReadableFormat": formats["markdown"],
-        "EntryContext": ec
+        if match_exact:
+            if ignore_case:
+                if not item.lower() in [list_item.lower().strip() for list_item in lst]:
```
Comment (yaakovi) [inline Scripts/FilterByList/FilterByList.py]: calculate this outside the loop: `[list_item.lower().strip() for list_item in lst]`
better performance.

File: Scripts/FilterByList/FilterByList.py [hunk 5]
```diff
@@ -2,64 +2,97 @@
...
-if isError(lst[0]):
-    demisto.results(lst)
-    sys.exit(0)
+def empty_list_context(lst, items, list_name):
+    not_white_listed = []  # type: list
+
+    for item in items:
+        not_white_listed.append(item)
+
+    ec = {'List.In': [], 'List.NotIn': not_white_listed}
```
Comment (yaakovi) [inline Scripts/FilterByList/FilterByList.py]: `ec = {'List.In': [], 'List.NotIn': items}`

File: Scripts/FilterByList/FilterByList.py [hunk 6]
```diff
@@ -90,9 +73,10 @@ def main():
     if not isinstance(items, list):
         items = items.split(',')
```
Comment (yaakovi) [inline Scripts/FilterByList/FilterByList.py]: should have been removed (this is part of the `argToList`)

**Pull request merged**